### PR TITLE
 PYTHON-71 - use prepared metadata and skip row result meta

### PR DESCRIPTION
**Pull request by aholmberg**
in protocol v2+

### CODE REVIEW

File: cassandra/query.py [hunk 1]
```diff
@@ -362,36 +361,34 @@ class PreparedStatement(object):
     may affect performance (as the operation requires a network roundtrip).
     """
 
-    column_metadata = None
+    column_metadata = None  #TODO: make this bind_metadata in next major
```
Comment (aboudreault) [inline cassandra/query.py]: added in PYTHON-504

File: cassandra/query.py [hunk 2]
```diff
@@ -362,36 +361,34 @@ class PreparedStatement(object):
...
+    consistency_level = None
+    custom_payload = None
+    fetch_size = FETCH_SIZE_UNSET
+    keyspace = None  # change to prepared_keyspace in major release
```
Comment (aboudreault) [inline cassandra/query.py]: added in PYTHON-504

File: cassandra/cluster.py [hunk 3]
```diff
@@ -2124,14 +2124,14 @@ def prepare(self, query, custom_payload=None):
         future = ResponseFuture(self, message, query=None, timeout=self.default_timeout)
         try:
             future.send_request()
-            query_id, column_metadata, pk_indexes = future.result()
+            query_id, bind_metadata, pk_indexes, result_metadata = future.result()
```
Comment (aboudreault) [inline cassandra/cluster.py]: In the future, if we extend that result part again, we might prefer to create a structure for the future result attributes.

**Comment by aboudreault** [comment]
LGTM!

File: cassandra/cluster.py (hunk 3 above)
Comment (aholmberg) [inline cassandra/cluster.py]: good idea. There are several places where this would be useful.

Comment (aholmberg) [inline cassandra/cluster.py]: @aboudreault I created https://datastax-oss.atlassian.net/browse/PYTHON-601 to refactor at some point. We might be able to work it into this release if there is time.

**Pull request merged**